syntax = "proto3";

service Dictionary {

  rpc About (EmptyRequest) returns (LineReply) {}

  rpc Author (EmptyRequest) returns (LineReply) {}

  rpc AllConcepts (EmptyRequest) returns (AllConceptsReply) {}

  rpc List (EmptyRequest) returns (ListReply) {}

  rpc Definition (DefinitionRequest) returns (DefinitionReply) {}

  rpc Create (CreateRequest) returns (CompleteReply) {}

  rpc Update (CreateRequest) returns (CompleteReply) {}

  rpc Remove (DefinitionRequest) returns (CompleteReply) {}

}

message EmptyRequest {}

message LineReply {
    string text = 1;
}

message AllConceptsReply {
    map<string, string> definitions = 1;
}

message ListReply {
    map<int32, string> concepts = 1;
}

message DefinitionRequest {
    string name = 1;
}

message DefinitionReply {
    string description = 1;
}

message CreateRequest {
    string name = 1;
    string description = 2;
}

message CompleteReply {
    int32 status = 1;
    string message = 2;
}


